### ISSUE TITLE
"XXX.fastp.fastq.gz" does not exist or is not a Bowtie 2 index"

### ISSUE THREAD

**Issue by suburbianmantia** [nf-core/smrnaseq#204]
Hi there!

I am trying to run the smrnaseq pipeline with the following command : 

```
nextflow run nf-core/smrnaseq \
--input /home/***/fastq_files/essai_2/Fastq_files_quater/samplesheet.csv -profile docker 
--outdir /home/***/fastq_files/ --multiqc_title ***
 --bt_indices /home/***/fastq_files/genome_bowtie2/  //here is the directory with bt2 index files
--save_reference true 
-c /home/***/***.config 
-resume
```

I think there is an issue with the fasta files I am using since it always fails with a contaminant fasta files that I downloaded from ensembl. 

The full error message is : 

```
Error executing process > 'NFCORE_SMRNASEQ:SMRNASEQ:CONTAMINANT_FILTER:MAP_CDNA (3)'

Caused by:
  Process `NFCORE_SMRNASEQ:SMRNASEQ:CONTAMINANT_FILTER:MAP_CDNA (3)` terminated with an error exit status (255)

Command executed:

  INDEX=`find -L ./ -name "*.3.ebwt" | sed 's/.3.ebwt//'`
  bowtie2 \
      --threads 6 \
      --very-sensitive-local \
      -k 1 \
      -x $INDEX \
      --un ADULT_3_P2.cDNA.filter.unmapped.contaminant.fastq \
      ADULT_3_P2.fastp.fastq.gz \
      [] \
      -S ADULT_3_P2.filter.contaminant.sam > ADULT_3_P2.contaminant_bowtie.log 2>&1

  # extracting number of reads from bowtie logs
  awk -v type=cDNA 'BEGIN{tot=0} {if(NR==4 || NR == 5){tot += $1}} END {print "\""type"\": "tot }' ADULT_3_P2.contaminant_bowtie.log | tr -d , > filtered.ADULT_3_P2_cDNA.stats

  cat <<-END_VERSIONS > versions.yml
  "NFCORE_SMRNASEQ:SMRNASEQ:CONTAMINANT_FILTER:MAP_CDNA":
      bowtie2: $(echo $(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*$//' | tr -d '')
  END_VERSIONS

Command exit status:
  255

Command output:
  (empty)

Work dir:
  /home/***/work/71/18c64715a17783f07c5fdf5df0c699

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`


Failed to invoke `workflow.onComplete` event handler

 -- Check script '/root/.nextflow/assets/nf-core/smrnaseq/./workflows/smrnaseq.nf' at line: 249 or see '.nextflow.log' file for more details
```

Also, the bowtie.log files states : 

```
"ADULT_3_P2.fastp.fastq.gz" does not exist or is not a Bowtie 2 index
```

but with the following files in the directory : 
ADULT_3_P2.fastp.fastq.gz  fasta_bidx.1.bt2  fasta_bidx.2.bt2  fasta_bidx.3.bt2  fasta_bidx.4.bt2  fasta_bidx.rev.1.bt2  fasta_bidx.rev.2.bt2 (with bt2 indices).

Any ideas of what could be the problem here?

Sorry for asking but I am a newbie in smrnaseq pipelines and nextflow

Thanks for helping!

**Comment by apeltzer** [comment]
One question: Why do you use some custom contaminants in the first place? Might be better to give things a start with a more default setting, then moving on more advanced usage. If you are already considering that the contaminatns you are using cause trouble, might not be the best to start with them ;-) 


```
 --bt_indices /home/***/fastq_files/genome_bowtie2/  //here is the directory with bt2 index files
--save_reference true 
```

It should be fine to use a prebuild index, but please remove the save_reference then - this is only necessary when you build everything from scratch. How did you build the indices btw? 

Another point:

```
-c /home/***/***.config 
```

Whats in there? Anything you set to custom stuff?

**Comment by suburbianmantia** [comment]
Hi there! 

First, sorry for being this long to respond and thank you for your response.

I was using the contaminant fasta files that were suggested on the nf-core/smrnaseq page actually. I removed all these fasta files and all went fine in the end. 

I built the indices with bowtie 1 instead of bowtie 2 in the first place and I assume that contributed to solve the issue.

Also : the config file was a custom config file that I made where I have the path to the different fasta files and some other parameters that I wanted to set up myself.

Anyway thanks a lot for the response, but for now I have no idea on how to filter contaminants with the pipeline since when I set a fasta file to filter contamination, the pipeline does not work on my local machine. So any idea on how to solve this issue?

Thanks a lot for your response!
